### PR TITLE
User life cycle hooks

### PR DESCRIPTION
**Pull request by patricegautier**
This is a continuation of 'Add a user option for lifecycle hooks #1098', which automatically got closed as I was doing some branch maintenance on my fork, sorry for the noise..

### CODE REVIEW

**Comment by patricegautier** [comment]
as a little bonus the test that was failing is fixed.

**Comment by patricegautier** [comment]
.. and I also added a post-update timeout option, symmetrical to the pre-update timeout one..

**Comment by piksel** [comment]
I refactored the container client tests to make it possible for better testing of this (using ghttp).

> .. and I also added a post-update timeout option, symmetrical to the pre-update timeout one..

This might be a good PR on it's own, but it should probably be separated. What was the motivation for adding it here?

**Comment by patricegautier** [comment]
> I refactored the container client tests to make it possible for better testing of this (using ghttp).

Thanks I will take a look.

> .. and I also added a post-update timeout option, symmetrical to the pre-update timeout one..
> 
> This might be a good PR on it's own, but it should probably be separated. What was the motivation for adding it here?

You are right it should have been separate.  Let me know if that's a must now that it's in there..

**Comment by piksel** [comment]
> Let me know if that's a must now that it's in there..

That part will need some discussion as well and having it in the same PR thread will get messy. And also, every PR gets turned into a change list item, and "User life cycle hooks" gives no indication about timeouts (and should be it's own item).

**Comment by patricegautier** [comment]
Ok no worries, I’ll take it out and create a separate PR.

	-P

**Comment by patricegautier** [comment]
FYI - removed the post update timeout bit - I will submit a separate PR

**Comment by patricegautier** [comment]
and post update timeout changes are now in #1124.  Thanks.

File: docs/lifecycle-hooks.md [hunk 1]
```diff
@@ -50,13 +50,19 @@ the `docker run` command line:
...
+com.centurylinklabs.watchtower.lifecycle.pre-check.user="myuser"
+
 ### Timeouts
 The timeout for all lifecycle commands is 60 seconds. After that, a timeout will
 occur, forcing Watchtower to continue the update loop.
 
 #### Pre-update timeouts
 
-For the `pre-update` lifecycle command, it is possible to override this timeout to
+For the `pre-update` the lifecycle command, it is possible to override this timeout to
```
Comment (piksel) [inline docs/lifecycle-hooks.md]:
```suggestion
For the `pre-update` lifecycle command, it is possible to override this timeout to
```

**Review by piksel — APPROVED** [review]
The docs revert accidently a word, but otherwise this looks good imo.

File: pkg/container/container_test.go [hunk 2]
```diff
@@ -213,6 +213,28 @@ var _ = Describe("the container", func() {
 					Expect(links).To(SatisfyAll(ContainElement("redis"), ContainElement("postgres"), HaveLen(2)))
 				})
 			})
+			When("there is a pre or post update timeout", func() {
+				It("should return minute values", func() {
```
Comment (simonaronsson) [inline pkg/container/container_test.go:217]:
```
				It("should return minute values", func() {
```

copy/paste artifact?

**Review by simonaronsson — CHANGES_REQUESTED** [review]

File: pkg/container/container_test.go (hunk 2 above)
Comment (piksel) [reply pkg/container/container_test.go:217]: it applies to one of the env variables, but it should probably be split into 
```suggestion
				It("should return the minute value", func() {
```
and 
```go
			When("there is a pre or post update user", func() {
				It("should return the specified user", func() {
```

**Pull request closed without merging**